### PR TITLE
Add image for opengraph embed

### PR DESCRIPTION
**Pull request by mkst**
So, some of the site icons do not play nicely with `vercel/og` and need fixing up:
- [x] PS2 (text on image)
- [x] Saturn (text on image)
- [x] 3DS (incorrect aspect ratio)

### CODE REVIEW

File: frontend/src/app/scratch/[slug]/opengraph-image.tsx [hunk 1]
```diff
@@ -13,67 +13,71 @@ export default async function ScratchOG({ params }: { params: { slug: string }})
...
 
     const score = getScoreText(compilation?.diff_output?.current_score ?? -1, compilation?.diff_output?.max_score ?? -1, scratch.match_override)
-    const fork = parentScratch ? `${parentScratch.owner?.username ?? "?"}/${parentScratch.name}` : ""
 
     const compilers = await get("/compiler")
     const preset = compilers["platforms"][scratch.platform].presets.find(p => p.id === scratch.preset)
 
+    // TODO: tune these
+    const scratchOwnerSize = scratch.owner.username.length > 16 ? "text-6xl" : "text-7xl"
+    const scratchNameSize = scratch.name.length > 20 ? "text-5xl" : "text-6xl"
```
Comment (bates64) [inline frontend/src/app/scratch/[slug]/opengraph-image.tsx:22]: I reckon name should be bigger than owner

File: frontend/src/app/scratch/[slug]/opengraph-image.tsx [hunk 2]
```diff
@@ -13,67 +13,71 @@ export default async function ScratchOG({ params }: { params: { slug: string }})
...
-                <PlatformIcon
-                    platform={scratch.platform}
-                    size={120}
-                />
+            <div tw="flex ml-20 mr-20 mt-20 flow-row justify-between">
                 <div tw="flex flex-col justify-center">
-                    <div tw="flex "></div>
-                    <div tw="flex ml-10 ">{preset?.name || "Custom Preset"}</div>
+                    <div tw={`flex ${scratchOwnerSize}`}>{scratch.owner.username}/</div>
+                    <div tw={`flex ${scratchNameSize}`}>{scratch.name}</div>
```
Comment (bates64) [inline frontend/src/app/scratch/[slug]/opengraph-image.tsx:32]: Swap these? And make the owner a darker colour?

**Review by bates64 — CHANGES_REQUESTED** [review]

File: frontend/src/app/scratch/[slug]/opengraph-image.tsx [hunk 3]
```diff
@@ -29,6 +29,10 @@ export default async function ScratchOG({ params }: { params: { slug: string }})
     const score = compilation?.diff_output?.current_score ?? -1
     const maxScore = compilation?.diff_output?.max_score ?? -1
 
+    const percent = scratch.match_override ? 100 : calculateScorePercent(score, maxScore)
+    const doneWidth = Math.floor(percent * 300 / 100) // Unsure why this is 300 rather than 1200...
+    const todoWidth = 300 - doneWidth
```
Comment (bates64) [inline frontend/src/app/scratch/[slug]/opengraph-image.tsx:35]: 300 should be made into a const and used as the image width too 
And changed to 1200 with the other change I'm suggesting

File: frontend/src/app/scratch/[slug]/opengraph-image.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+                </div>
+
+                <div tw="flex flex-col justify-around ml-15">
+                    <PurpleFrog width={64} height={64} />
+                </div>
+
+            </div>
+            <div tw="flex">
+                <div tw={`flex h-4 bg-purple-500 w-${doneWidth}`}></div>
+                <div tw={`flex h-4 bg-purple-900 w-${todoWidth}`}></div>
```
Comment (bates64) [inline frontend/src/app/scratch/[slug]/opengraph-image.tsx:114]: `w-300` in Tailwind doesn't correspond to 300 pixels, it instead corresponds to a special scale: https://tailwindcss.com/docs/width

Instead, these should be [arbitrary values](https://tailwindcss.com/docs/width#arbitrary-values):
```suggestion
                <div tw={`flex h-4 bg-purple-500 w-[${doneWidth}`}px]></div>
                <div tw={`flex h-4 bg-purple-900 w-[${todoWidth}`}px]></div>
```

File: frontend/src/app/scratch/[slug]/opengraph-image.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            </div>
+
+        </div>,
+        {
+            width: 1200,
```
Comment (bates64) [inline frontend/src/app/scratch/[slug]/opengraph-image.tsx:119]: Variable

File: frontend/src/app/scratch/[slug]/opengraph-image.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+            height: 400,
+        },
+    )
+}
```
Comment (bates64) [inline frontend/src/app/scratch/[slug]/opengraph-image.tsx:123]: Missing newline at eof

**Review by bates64 — CHANGES_REQUESTED** [review]

File: frontend/src/app/scratch/[slug]/opengraph-image.tsx (hunk 4 above)
Comment (mkst) [reply frontend/src/app/scratch/[slug]/opengraph-image.tsx]: Makes sense, thanks for all the feedback, have pushed a fresh commit!

**Pull request merged**